File: Esercitazioni/Esercitazione0/svolta/consumatore.c
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#define MAX_STRING_LENGTH 256

// consumatore.c e' un filtro
int main(int argc, char *argv[]) {

    char *file_in, read_char, buf[MAX_STRING_LENGTH];
    int   nread, fd;

    // controllo numero argomenti
    if (argc != 2) {
        perror(" numero di argomenti sbagliato");
        exit(1);
    }

    file_in = argv[1];
    fd      = open(file_in, O_RDONLY);
    if (fd < 0) {
        perror("P0: Impossibile aprire il file.");
        exit(2);
    }

    while (nread = read(fd, &read_char, sizeof(char))) /* Fino ad EOF*/ {
        if (nread > 0)
            putchar(read_char);
        else {
            printf("(PID %d) impossibile leggere dal file %s", getpid(), file_in);
            perror("Errore in lettura");
            close(fd);
            exit(3);
        }
    }
    close(fd);
}

File: Esercitazioni_NEW/Esercitazione0/Codice/Svolta/Consumatore.java
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;

// Consumatore e' un filtro
public class Consumatore {
	public static void main(String[] args) {
		FileReader r = null;
		char ch;
		int x;
		
		if (args.length != 1){
			System.out.println("Utilizzo: consumatore <inputFilename>");
			System.exit(0);
		}
	  
		try {
			r = new FileReader(args[0]);
		} catch(FileNotFoundException e){
			System.out.println("File non trovato");
			System.exit(1);
		}
		try {

			while ((x = r.read()) >= 0) { 
				ch = (char) x;
				System.out.print(ch);
			}
			r.close();
		} catch(IOException ex){
			System.out.println("Errore di input");
			System.exit(2);
		}
}}
